src/opm/parser/eclipse/EclipseState/Schedule/ArrayDimChecker.cpp

```diff
@@ -41,7 +41,7 @@ namespace {
                            const Opm::ParseContext& ctxt,
                            Opm::ErrorGuard&         guard)
         {
-            const auto nWells = sched.numWells();
+            auto nWells = sched.numWells();
```

Comment:
This seems like a step backwards.  Does GCC 8.3 really complain about declaring an object of built-in type (here `size_t`) as `const`?

Reply:
It actually complains in [line 46](/OPM/opm-common/blob/393a543ed5665dab479857671ee7f282ff1f28bb/src/opm/parser/eclipse/EclipseState/Schedule/ArrayDimChecker.cpp#L46) but doing `if (nWells > static_cast<typename std::remove_const<decltype(nWells)>::type>(wdims.maxWellsInField()))` felt a bit over the top.

Reply:
How about
```C++
if (nWells > decltype(nWells){dims.maxWellsInField()})
```
instead?

Also, what was the compiler diagnostic for this.  I'd be really interested in seeing it.